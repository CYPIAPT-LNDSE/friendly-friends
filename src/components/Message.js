import React from 'react';
import ResponseOption from './ResponseOption.js';

class Message extends React.Component {
  constructor(props) {
    super(props);
    this.updateMessage = this.updateMessage.bind(this);
  }
  updateMessage(next) {
    // TODO - dispatch action here
  }
  render() {
    const { message } = this.props;
    return (
      <div className="message">
        <p>{message.text}</p>
        {message.responses.map((response, i) =>
          <ResponseOption updateMessage={this.updateMessage}
                          response={response}
                          key={i} />
        )}
      </div>
    );
  }
}

export default Message;
