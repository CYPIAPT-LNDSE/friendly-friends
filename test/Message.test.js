import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../src/components/Message.js';

test('Message renders correctly', () => {
  const tree = renderer.create(<Message/>).toJSON();
  expect(tree).toMatchSnapshot();
});
