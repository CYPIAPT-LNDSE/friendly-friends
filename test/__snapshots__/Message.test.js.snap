exports[`test Message renders correctly 1`] = `
<div
  className="message">
  <p>
    Hi! how are you feeling
  </p>
  <p
    className="response-option"
    onClick={[Function]}>
    I feel sad
  </p>
  <p
    className="response-option"
    onClick={[Function]}>
    I am happy!
  </p>
</div>
`;
